packages/core/ios/Sources/DdSdkNativeInitialization.swift

```diff
@@ -0,0 +1,225 @@
+        DatadogSDKWrapper.shared.enableLogs(with: logsConfig)
+        
+        let traceConfig = buildTraceConfiguration(configuration: sdkConfiguration)
+        DatadogSDKWrapper.shared.enableTrace(with: traceConfig)
+
+        if sdkConfiguration.nativeCrashReportEnabled ?? false {
+            DatadogSDKWrapper.shared.enableCrashReporting()
+        }
+        
+        DatadogSDKWrapper.shared.enableWebviewTracking()
+    }
+
+    func buildSDKConfiguration(configuration: DdSdkConfiguration, defaultAppVersion: String = getDefaultAppVersion()) -> Datadog.Configuration {
+        var config = Datadog.Configuration(
+            clientToken: configuration.clientToken,
+            env: configuration.env,
+            site: configuration.site,
+            service: configuration.serviceName as? String,
+            batchSize: configuration.batchSize,
+            uploadFrequency: configuration.uploadFrequency,
+            proxyConfiguration: configuration.proxyConfig
+        )
+
+        if var additionalConfiguration = configuration.additionalConfig as? [String: Any] {
+            if let versionSuffix = additionalConfiguration[InternalConfigurationAttributes.versionSuffix] as? String {
+                let datadogVersion = defaultAppVersion + versionSuffix
+                additionalConfiguration[CrossPlatformAttributes.version] = datadogVersion
+            }
+
+            config._internal_mutation {
+              $0.additionalConfiguration = additionalConfiguration
+            }
+        }
+
+        return config
+    }
+    
+    func buildRUMConfiguration(configuration: DdSdkConfiguration) -> RUM.Configuration {
+        var longTaskThreshold: TimeInterval? = nil
+        if let threshold = configuration.nativeLongTaskThresholdMs {
+            if (threshold != 0) {
+                // `nativeLongTaskThresholdMs` attribute is in milliseconds
+                longTaskThreshold = threshold / 1_000
+            }
+        }
+        
+        var uiKitViewsPredicate: UIKitRUMViewsPredicate? = nil
+        if let enableViewTracking = configuration.nativeViewTracking, enableViewTracking {
+            uiKitViewsPredicate = DefaultUIKitRUMViewsPredicate()
+        }
+
+        var uiKitActionsPredicate: UIKitRUMActionsPredicate? = nil
+        if let enableInteractionTracking = configuration.nativeInteractionTracking, enableInteractionTracking {
+            uiKitActionsPredicate = DefaultUIKitRUMActionsPredicate()
+        }
+        
+        var urlSessionTracking: RUM.Configuration.URLSessionTracking? = nil
+        if let firstPartyHosts = configuration.firstPartyHosts {
+            // We will always fall under this condition as firstPartyHosts is an empty array by default
+            urlSessionTracking = RUM.Configuration.URLSessionTracking(
+                firstPartyHostsTracing: .traceWithHeaders(
+                    hostsWithHeaders: firstPartyHosts,
+                    sampleRate: 100.0
```

Comment:
Is this right? Sample rate can't be configured?

Reply:
This part is copy/pasted from the previous implementation. The way I remember it, we already apply a sampling rate on the JS layer when adding the header to the request. We need resources to be tagged both on the JS (for the backend to get the trace id) and on native layer (for RUM to receive the trace id on the resource).
If we would apply sampling on both sides, we would end up with less correlated traces.

However I'll double check here and add a comment to explain in any case.

Reply:
I've run some tests and this snippet does not seem to change anything for RUM Resources that were created on the JS side.
I think there is some investigation that is needed before we can safely remove it, I might have missed something and it would require to understand how RUM correlation with traces works on the iOS SDK.

I'll make a change to use the sampling rate specified in the config so native RUM resources get the correct ones.